invenio_app_ils/documents/covers_builder.py

```diff
@@ -7,14 +7,37 @@
 
 """Document configuration callbacks."""
 
+from flask import url_for
+
 
 def build_ils_demo_cover_urls(metadata):
     """Build working ulrs for demo data."""
-    default_url = "http://covers.openlibrary.org/b/isbn"
-    cover_meta = metadata.get("cover_metadata", {})
-    isbn = cover_meta.get("isbn", "")
+    cover_metadata = metadata.get("cover_metadata", {})
+    isbn = cover_metadata.get("ISBN", "")
+    if isbn:
+        return build_openlibrary_urls(isbn)
+    return build_placeholder_urls()
+
+
+def build_openlibrary_urls(isbn):
+    """Build Open Library urls."""
+    url = "http://covers.openlibrary.org/b/isbn"
+    return {
+        "small": "{url}/{isbn}-S.jpg".format(url=url, isbn=isbn),
+        "medium": "{url}/{isbn}-M.jpg".format(url=url, isbn=isbn),
+        "large": "{url}/{isbn}-L.jpg".format(url=url, isbn=isbn),
+    }
+
+
+def build_placeholder_urls():
+    """Build urls for default cover placeholders."""
+    image_path = url_for(
```

Comment:
I am hesitant on placing this to be served by the backend. Because we mix now frontend and backend, especially if we will overwrite, everything on the frontend will be overwritten in react or theme and just this one on the backend. Plus we will operate on split repos... @ntarocco WDYT?

Reply:
This is actually the reason I re-worked on it. Initially I used the existing placeholder that was in ui public, but as @ntarocco foresaw the end client might be anything from webapp, mobile app or just a rest consumer, and then there will be missing a cover.

Reply:
It is a long discussion, and we had many with Chris, here the TL;DR:
* ILS should not know what is your book provider: it could be Invenio itself (e.g. a folder with all the covers), or an external book provider. By default, it is the first, and so the backend should generate the URLs, based on the record fields
* The frontend should only put a `<img` with the URL, nothing else. If you start to have logic in the frontend, you might end up with this mixing part of the logic in the backend, to prepare some data, and part in the frontend
* The default place holder could be provided by the book cover provider or by your backend, and that's why we decided to go for this solution.

We can discuss IRL if you want.

Reply:
I discussed with @ntarocco, to solve possible UI problems I was thinking about,  if you attach in the response payload for the cover a flag that it indicates we are sending the placeholder url, I think it would be ok like `no_cover: True` so the UI can know that it is getting a placeholder.

Reply:
Done, very good idea, it is `is_placeholder` the property that comes in `cover_metadata`.